### PR TITLE
Feature/replace buefy autocomplete

### PR DESCRIPTION
**Pull request by hannyle**
### Description

Replace Buefy Autocomplete component to csc-ui component.

### Related issues
GL issue 944

### Type of change

- [x] New feature (non-breaking change which adds functionality)

### Changes Made

- My prior PR for adding possibility to have custom menu items in [csc-ui repo](https://github.com/CSCfi/csc-ui/pull/190)
- Since that PR is still waiting for review and not yet merged to csc-ui `master` branch, I used [my forked repo](https://github.com/hannyle/csc-ui/tree/master) from csc-ui to continue developing our own Autcomplete component
- Updated Changelog

### Testing

- [x] Manual testing

### CODE REVIEW

File: swift_browser_ui_frontend/src/components/SearchBox.vue [hunk 1]
```diff
@@ -3,46 +3,41 @@
...
-      :data="searchResults"
-      field="name"
-      :open-on-focus="true"
-      :keep-first="true"
-      :loading="isSearching"
-      max-height="350px"
-      @select="option => $router.push(getSearchRoute(option))"
-      @focus="event => searchGainedFocus()"
+      hide-details
+      required
```
Comment (csc-felipe) [inline swift_browser_ui_frontend/src/components/SearchBox.vue:14]: why does it have `required`?

**Review by csc-felipe — APPROVED** [review]
lgtm, well done

File: swift_browser_ui_frontend/src/components/SearchBox.vue [hunk 2]
```diff
@@ -222,8 +226,14 @@ export default {
...
+ @import "@/css/prod.scss";
 .empty-search {
   height: 2rem;
 }
+
+div[aria-selected='true'], div[slot="customMenu"]:hover {
+  background-color: $csc-primary-lighter;
+}
+
 </style>
```
Comment (sampsapenna) [inline swift_browser_ui_frontend/src/components/SearchBox.vue:239]: Missing trailing newline

**Review by sampsapenna — DISMISSED** [review]

File: swift_browser_ui_frontend/src/components/SearchBox.vue (hunk 1 above)
Comment (hannyle) [reply swift_browser_ui_frontend/src/components/SearchBox.vue:14]: good catch. It shouldn't be required!

**Review by csc-felipe — APPROVED** [review]

**Review by sampsapenna — APPROVED** [review]

**Comment by hannyle** [comment]
Is it okay if I merge this PR with **csc-ui forked repo** as dependency? We could change it back to the correct version when the design system team merges my PR?

**Comment by csc-felipe** [comment]
Yes, let's do it like that

**Pull request merged**